docs/core/common/assets/js/api/core/commands.md

```diff
@@ -0,0 +1,174 @@
+    * Commands data - Base class: `$e.modules.CommandData` - `[DATA]` commands for communicate with the _data\cache\backend_.
+  * Each [component](../core/components.md#guidelines-conventions--files-structure), can override few methods `defaultCommands`, `defaultCommandsInternal`, `defaultData` 
+  methods which are used to import the command(s), according to their type.
+  * The commands should be imported via built-in method called `importCommands`.
+    * Example:
+        ```html class:"lineNo"
+        1  📦 component
+        2  │   📜 component.js
+        3  │
+        4  └───📂 commands
+        5  │   │   📜 index.js ( has all the commands exported )
+        6  │   │   📜 exmaple-command.js
+        7  │   │   ...
+        ```
+        `commands/index.js` file at line *5*:
+        ```javascript
+        export { ExmapleCommand } from './example-command';
+        ```
+    * use `importCommands` example: `component.js` file at line *2*:
+        ```javascript
+        import * as commands from './commands/';
+
+        export class Component extends $e.modules.ComponentBase {
+            getNamespace() {
+                return 'component-name';
+            }
+
+            defaultCommands() {
+                return this.importCommands( commands );
+            }
+        }
+        ```
+  * All series of commands type should have unique folder and index file to hold them:
+    * Commands:
+        * As the example above.
+    * Commands internal
+        * Example:
+            ```html class:"lineNo"
+            1  📦 component
+            2  │   📜 component.js
+            3  │
+            4  └───📂 commands-internal
+            5  │   │   📜 index.js ( has all the commands exported )
+            6  │   │   📜 internal-command.js
+            7  │   │   ...
+            ```
+        * `commands-internal/index.js` file at line *5*:
+            ```javascript
+            export { InternalCommand } from './internal-command';
+            ```
+        * use `importCommands` example: `component.js` file at line *2*:
+            ```javascript
+            import * as commandsInternal from './commands-internal/';
+    
+            export class Component extends $e.modules.ComponentBase {
+                getNamespace() {
+                    return 'component-name';
+                }
+    
+                defaultCommandsInternal() {
+                    return this.importCommands( commandsInternal );
+                }
+            }
+            ```
+    * Commands data
+        * Example:
+            ```html class:"lineNo"
+            1  📦 component
+            2  │   📜 component.js
+            3  │
+            4  └───📂 commands-data
+            5  │   │   📜 index.js ( has all the commands exported )
+            6  │   │   📜 data-command.js
+            7  │   │   ...
+            ```
+        * `index.js` file at line *5*:
```

Comment:
Add the dir name for all `index.js`. here and next.

Reply:
Handled.